[MRG] Pin cython version until cython issue 1790 is resolved

betatim opened the pull request:
See discussion here: https://github.com/cython/cython/issues/1790

I propose we pin the version of cython for the time being. Sounds like this is a bug in cython -> behaviour in 0.25.2 is correct and will be restored in a future version.

- [x] Is it mergeable?
- [ ] `make test` Did it pass the tests?

betatim committed:
Pin cython version until cython issue 1790 is resolved

See discussion here: https://github.com/cython/cython/issues/1790

ctb commented:
+1

camillescott reviewed (dismissed):
seems like a fine temp workaround to me

betatim commented:
[This line](https://github.com/dib-lab/khmer/blob/master/Makefile#L105) causes an upgrade of cython to 0.26. [A bit of pip logic explained](https://pip.pypa.io/en/stable/user_guide/#only-if-needed-recursive-upgrade), so I think we would have to install all the dependencies explicitly in the CI environment and not call `make dep`? A bit suboptimal

camillescott commented:
Seems to me like we shouldn't be upgrading packages for folks -- just
giving pip the reqs file without `--upgrade` should only upgrade the
packages that need it, no?

On Jul 22, 2017 11:29 PM, "Tim Head" <notifications@github.com> wrote:

> This line <https://github.com/dib-lab/khmer/blob/master/Makefile#L105>
> causes an upgrade of cython to 0.26. A bit of pip logic explained
> <https://pip.pypa.io/en/stable/user_guide/#only-if-needed-recursive-upgrade>,
> so I think we would have to install all the dependencies explicitly in the
> CI environment and not call make dep? A bit suboptimal
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/dib-lab/khmer/pull/1745#issuecomment-317232500>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/ACwxrd3lhQMLYrcEKuUxWjHkr5uWPnpuks5sQuhFgaJpZM4OfdV6>
> .
>

betatim commented:
As usual, I assume that `--upgrade` and `--ignore-installed` were put there for a reason...

I think without `--upgrade` `pip install -r req.txt` will update your foobar package from vX (installed on your system) to vY (minimum required by khmer) but won't move it all the way to vZ (latest).

So I think we should we should drop `--upgrade --ignore-installed`, but see first line of this comment :)

betatim committed:
Remove forced upgrade and ignoring of user packages

`make install-deps` does not need to force an upgrade or ignore
packages the user already has installed.

betatim committed:
Pin cython in docs/requirements.txt as well

betatim commented:
Do we really need cython as a requirement in `docs/requirements.txt`? That seems weird no?

standage commented:
Yes, I was troubleshooting broken ReadTheDocs builds. See https://github.com/dib-lab/khmer/issues/1733#issuecomment-308496816.

I added it only reluctantly in the first place, would be happy to remove it if it's no longer a problem or if we find an alternative solution.

betatim commented:
Aha! I will pin it as well for the moment as I don't have a smart idea about RTD :-/

camillescott reviewed (approved):
looks all well and good to me now
